src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -79,16 +79,28 @@ public AWSLoggerCore(AWSLoggerConfig config, string logType)
             _config = config;
             _logType = logType;
 
-            var credentials = DetermineCredentials(config);
-
-            if (_config.Region != null)
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials, Amazon.RegionEndpoint.GetBySystemName(_config.Region));
-            }
-            else
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials);
-            }
+            if (!string.IsNullOrEmpty(_config.ServiceUrl))
+            {
+                var awsConfig = new AmazonCloudWatchLogsConfig() { ServiceURL = _config.ServiceUrl };
+                if (_config.ServiceUrl.StartsWith("http://", StringComparison.OrdinalIgnoreCase))
+                    awsConfig.UseHttp = true;
+                if (_config.Credentials != null)

```

Comment:
Please add a space before starting this condition on credentials. I first read this as one continually conditional logic series.

Reply:
Fixed